modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/property/wrappertype/StackedType.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.java.property.wrappertype;
+
+import static java.util.Optional.ofNullable;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ValueResolver;
+
+import java.util.Optional;
+
+/**
+ * Represents a certain type which is the generic wrapper of another one.
+ *
+ * @since 4.0
+ */
+public class StackedType {
+
+  private final Class type;
+  private final ExpressionBasedResolverFactory expressionBasedResolverFactory;
+  private final StaticResolverFactory staticResolverFactory;
+  private final DelegateResolverFactory delegateResolverFactory;
+
+  /**
+   * Creates a new instance.
+   * If a factory is not provided it will be considered that the correspondent type doesn't support {@link ValueResolver}
+   * of the factory type.
+   *
+   * @param type                           The represented type
+   * @param expressionBasedResolverFactory A factory that creates instances of expression based {@link ValueResolver value resolvers}
+   * @param staticResolverFactory          A factory that creates instances of static {@link ValueResolver value resolvers}
+   * @param delegateResolverFactory        A factory that create instances of {@link ValueResolver value resolver} wrappers
+   */
+  private StackedType(Class type, ExpressionBasedResolverFactory expressionBasedResolverFactory,
```

Comment:
still an unclear name. This is not representing a type, but a resolver of some kind

Reply:
this is representing a type and their resolvers, I think that is a pretty representable name in their context.